tests/unit/models/keras/test_architectures.py

```diff
@@ -262,3 +280,142 @@ def test_gaussian_network_is_correctly_constructed(
     assert isinstance(network_built.layers[0], tf.keras.layers.InputLayer)
     assert len(network_built.layers[1:-2]) == num_hidden_layers
     assert isinstance(network_built.layers[-1], tfp.layers.DistributionLambda)
+
+
+@pytest.mark.mcdropout
+@pytest.mark.parametrize("num_hidden_layers, rate", [(1, 0.3), (3, 0.7), (5, 0.9)])
+@pytest.mark.parametrize("units", [10, 50])
+@pytest.mark.parametrize("activation", ["relu", tf.keras.activations.tanh])
+def test_dropout_network_build_seems_correct(
+    dropout_network: DropoutNetwork,
+    query_point_shape: List[int],
+    observation_shape: List[int],
+    num_hidden_layers: int,
+    units: int,
+    activation: Union[str, tf.keras.layers.Activation],
+    rate: float,
+) -> None:
+    """Tests the correct consturction of dropout network architectures"""
+
+    example_data = empty_dataset(query_point_shape, observation_shape)
+    inputs, outputs = get_tensor_spec_from_data(example_data)
+    hidden_layer_args = [
+        {"units": units, "activation": activation} for _ in range(num_hidden_layers)
+    ]
+
+    dropout_nn = dropout_network(inputs, outputs, hidden_layer_args, rate)
+
+    if not isinstance(rate, list):
+        rate = [rate for _ in range(num_hidden_layers + 1)]
+
+    # basics
+    assert isinstance(dropout_nn, tf.keras.Model)
+
+    # check the model has not been compiled
+    assert dropout_nn.compiled_loss is None
+    assert dropout_nn.compiled_metrics is None
+    assert dropout_nn.optimizer is None
+
+    # check correct number of layers and proerply constructed
+    assert len(dropout_nn.layers) == 2
```

Comment:
should the value on the right depend on `num_hidden_layers` argument?

Reply:
No because I have a Sequential layer that contains all the hidden layers and then a Sequential layer that contains the output layers. Could be potentially simplified, but I thought it organized it nicely since typically all the hidden layers are the same and the output layer is different.